cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/cache/SqlCacheCleanupAgent.kt

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.cats.sql.cache
+
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE
+import com.netflix.spinnaker.cats.agent.CachingAgent
+import com.netflix.spinnaker.cats.agent.RunnableAgent
+import com.netflix.spinnaker.cats.provider.ProviderRegistry
+import com.netflix.spinnaker.clouddriver.cache.CustomScheduledAgent
+import com.netflix.spinnaker.clouddriver.core.provider.CoreProvider
+import com.netflix.spinnaker.config.ConnectionPools
+import com.netflix.spinnaker.kork.sql.routing.withPool
+import org.jooq.DSLContext
+import org.jooq.Field
+import org.jooq.impl.DSL.field
+import org.jooq.impl.DSL.table
+import org.slf4j.LoggerFactory
+import java.sql.SQLException
+import java.util.concurrent.TimeUnit
+
+/**
+ * Intermittently scans the entire database looking for records created by caching agents that
+ * are no longer configured.
+ */
+class SqlCacheCleanupAgent(
+  private val providerRegistry: ProviderRegistry,
+  private val jooq: DSLContext,
+  private val registry: Registry,
+  private val sqlNames: SqlNames
+) : RunnableAgent, CustomScheduledAgent {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  private val deletedId = registry.createId("sql.cacheCleanupAgent.dataTypeRecordsDeleted")
+  private val timingId = registry.createId("sql.cacheCleanupAgent.dataTypeCleanupDuration")
```

Comment:
Longer metric names. I'm often left having no other option but read code to know what some of our metric names actually mean (e.g. `sql.cacheCleanupAgent.deleted` would probably be a likely metric name elsewhere).